x-pack/plugins/ingest_manager/common/services/datasource_to_agent_datasource.ts

```diff
@@ -23,12 +24,26 @@ export const storedDatasourceToAgentDatasource = (
           if (stream.config) {
             const fullStream = {
               ...stream,
-              ...Object.entries(stream.config).reduce((acc, [configName, configValue]) => {
-                if (configValue !== undefined) {
-                  acc[configName] = configValue;
-                }
-                return acc;
-              }, {} as { [key: string]: any }),
+              ...Object.entries(stream.config).reduce(
+                (acc, [configName, { type: configType, value: configValue }]) => {
+                  if (configValue !== undefined) {
+                    if (configType === 'yaml') {
+                      try {
+                        const yamlValue = safeLoad(configValue);
+                        if (yamlValue) {
+                          acc[configName] = yamlValue;
+                        }
+                      } catch (e) {
+                        // Silently swallow parsing error
```

Comment:
Is this something we could send into debug logs (if there is such a thing?)

Reply:
[Kibana logging](https://github.com/elastic/kibana/blob/master/src/core/server/logging/README.md) is only on server-side. We should validate that the yaml can be parsed before allowing the user to continue to review step to avoid this branch, I have that captured in https://github.com/elastic/kibana/issues/60446